modules/extra/variables.tf

```diff
@@ -0,0 +1,94 @@
+variable "model" {
+  description = "The name of the Juju model to deploy to."
+  type        = string
+  default     = "identity-platform"
+}
+
+variable "deploy_extra_charms" {
+  type        = bool
+  default     = false
+  description = "Whether to deploy Identity Platform extensions: Admin UI, external IdP integrator, OpenFGA"
+}
+
+variable "openfga" {
+  description = "The configurations of the OpenFGA application."
+  type = object({
+    units   = optional(number, 1)
+    trust   = optional(bool, true)
+    config  = optional(map(string), {})
+    channel = optional(string, "latest/stable")
+    base    = optional(string, "ubuntu@22.04")
+  })
+  default = {}
+}
+
+variable "admin_ui" {
+  description = "The configurations of the Identity Platform Admin UI application."
+  type = object({
+    units   = optional(number, 1)
+    trust   = optional(bool, true)
+    config  = optional(map(string), {})
+    channel = optional(string, "latest/edge")
+    base    = optional(string, "ubuntu@22.04")
+  })
+  default = {}
+}
+
+variable "idp_provider_config" {
+  description = "The external IdP provider configurations."
+  type = object({
+    client_id : string
+    issuer_url : optional(string)
+    provider : string
+    provider_id : string
+    scope : optional(string, "profile email address phone")
+    microsoft_tenant_id : optional(string)
+    apple_team_id : optional(string)
+    apple_private_key_id : optional(string)
+  })
+  default = {
+    client_id   = "client_id"
+    provider    = "generic"
+    provider_id = "provider_id"
+  }
+}
+
+variable "idp_provider_credentials" {
+  description = "The external IdP provider credentials."
+  type = object({
+    client_secret : string
+    apple_private_key : optional(string)
+  })
+  default = {
+    client_secret = "client_secret"
+  }
+  sensitive = true
+}
+
+variable "ingress_offer_url" {
+  description = "Ingress Offer URL"
+  type        = string
+  default     = "admin/core.ingress"
+}
+
+variable "postgresql_offer_url" {
+  description = "PostgreSQL Offer URL"
+  type        = string
+  default     = "admin/core.postgresql"
+}
+
+variable "send_ca_certificate_offer_url" {
+  description = "Send CA Certificate Offer URL"
+  type        = string
+  default     = "admin/core.send-ca-cert"
+}
+
+variable "kratos_name" {
```

Comment:
This makes the assumption that the extra module will be deployed on the same model as kratos/hydra. We should at least document it somewhere (or make it work in the general case)

Reply:
added info in description

Reply:
https://github.com/canonical/iam-bundle-integration/blob/e6c2718d2057d72ba6c400158000065e6806fadb/modules/extra/variables.tf#L81

is this enough?